FootballStats/FootballStats/Repository/ScheduleRepository.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  ScheduleRepository.swift
+//  FootballStats
+//
+//  Created by Hoang Linh Nguyen on 4/10/2023.
+//
+
+import Foundation
+import Alamofire
+import RxSwift
+
+protocol ScheduleRepository {
+    func requestSchedule() -> Observable<[FixtureResponse]>
+}
+
+final class ScheduleRepositoryImpl: ScheduleRepository {
+    func requestSchedule() -> Observable<[FixtureResponse]> {
+        let footballHeaders: HTTPHeaders = [
+            "X-RapidAPI-Key": Constants.APIKey,
+            "X-RapidAPI-Host": Constants.APIFootballHost
+        ]
+        let footballParameters = ["season": Constants.season,
+                                  "league": Constants.leagueId]
+
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+        let currentDate = Date().getCurrentDate()
+        let lastweekDate = Date().getLastWeekDate()
+
+        let dateParameter = ["to": dateFormatter.string(from: currentDate),
+                             "from": dateFormatter.string(from: lastweekDate)]
+
+        let parameter = footballParameters.merging(dateParameter, uniquingKeysWith: +)
```

Comment:
```suggestion
        let footballParameters = ["season": Constants.season,
                                  "league": Constants.leagueId]

        let dateFormatter = DateFormatter()
        dateFormatter.dateFormat = "yyyy-MM-dd"
        let currentDate = Date().getCurrentDate()
        let lastweekDate = Date().getLastWeekDate()

        let dateParameter = ["to": dateFormatter.string(from: currentDate),
                             "from": dateFormatter.string(from: lastweekDate)]

        let parameter = footballParameters.merging(dateParameter, uniquingKeysWith: +)
```
Tách ra khai báo thành class `ScheduleRequest` đặt trong folder
`FootballStats/FootballStats/Data/Source/API/request`